### PR TITLE
Adding account purge operation

### PR DESCRIPTION
**Pull request by matthiashanel**
The new request is available for the system account.
The subject to send the request to is `$JS.API.ACCOUNT.PURGE.*`
With the name of the account to purge instead of the wildcard.

Also added directory cleanup code such that server do not
end up with empty streams directories and account dirs that
only contain streams

Also adding ACCOUNT to leaf node domain rewrite table

Addresses #3186 and #3306 by providing a way to
get rid of the streams for existing and non existing accounts

### CODE REVIEW

File: server/jetstream.go [hunk 1]
```diff
@@ -879,8 +902,13 @@ func (s *Server) shutdownJetStream() {
 			accounts = append(accounts, a)
 		}
 	}
+	accPurgeSub := js.accountPurge
 	js.mu.RUnlock()
 
+	if accPurgeSub != nil {
+		s.sysUnsubscribe(accPurgeSub)
```
Comment (derekcollison) [inline server/jetstream.go:909]: Should we set js.accountPurge to nil?

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -3074,6 +3094,24 @@ func (js *jetStream) processClusterDeleteStream(sa *streamAssignment, isMember,
 		sa.Group.node.Delete()
 	}
 
+	// Cleanup in case account does not exist or node was nil
+	if sacc := s.SystemAccount(); sacc != nil {
```
Comment (derekcollison) [inline server/jetstream_cluster.go:3098]: In the happy path this is always handled, this looks a bit strange here.
If the sa.Group.node.Delete() was called from above this is already handled etc.

File: server/jetstream_cluster.go [hunk 3]
```diff
@@ -3499,12 +3542,22 @@ func (js *jetStream) processClusterDeleteConsumer(ca *consumerAssignment, isMemb
 		ca.Group.node.Delete()
 	}
 
+	// Cleanup in case account does not exist or node was nil
+	if sacc := s.SystemAccount(); sacc != nil {
```
Comment (derekcollison) [inline server/jetstream_cluster.go:3546]: Again this looks out of place here, maybe we should have this to be pedantic to an extreme, but..

**Review by derekcollison — COMMENTED** [review]
I see we added in a purge subscription to capture these commands. Would we not want this to be a replicated meta state such that servers offline, once restarted would also perform the purge?

**Review by derekcollison — COMMENTED** [review]
Ignore comment about subscription vs meta state, got it now..

File: server/jetstream_cluster.go [hunk 4]
```diff
@@ -3094,8 +3094,10 @@ func (js *jetStream) processClusterDeleteStream(sa *streamAssignment, isMember,
 		sa.Group.node.Delete()
 	}
 
-	// Cleanup in case account does not exist or node was nil
-	if sacc := s.SystemAccount(); sacc != nil {
+	// This is a stop gap cleanup in case
+	// 1) the account does not exist (and mset couldn't be stopped) and/or
+	// 2) node was nil (and couldn't be deleted)
+	if sacc := s.SystemAccount(); sacc != nil && (!stopped || sa.Group.node == nil) {
```
Comment (derekcollison) [inline server/jetstream_cluster.go:3100]: Would reverse so in happy path do not need server lock to lookup system account. Also deref on sa.Group.node should be under js lock, so maybe capture above.

File: server/jetstream_cluster.go [hunk 5]
```diff
@@ -3542,8 +3546,10 @@ func (js *jetStream) processClusterDeleteConsumer(ca *consumerAssignment, isMemb
 		ca.Group.node.Delete()
 	}
 
-	// Cleanup in case account does not exist or node was nil
-	if sacc := s.SystemAccount(); sacc != nil {
+	// This is a stop gap cleanup in case
+	// 1) the account does not exist (and mset consumer couldn't be stopped) and/or
+	// 2) node was nil (and couldn't be deleted)
+	if sacc := s.SystemAccount(); sacc != nil && (!stopped || ca.Group.node == nil) {
```
Comment (derekcollison) [inline server/jetstream_cluster.go:3552]: Same here.

File: server/jetstream_cluster.go (hunk 4 above)
Comment (matthiashanel) [reply server/jetstream_cluster.go]: fixed (also modified earlier sa.Group.node access)

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**